aten/src/ATen/core/ivalue_inl.h

```diff
@@ -1194,6 +1202,14 @@ inline T IValue::to() && {
   return generic_to(std::move(*this), _fake_type<T>{});
 }
 
+template <>
+inline c10::optional<c10::string_view> IValue::to() && {
+  // In the default implementation, the IValue is destroyed with std::move.
+  // But if the unboxed type is optional<string_view> we cannot destroy
+  // the IValue.
+  return generic_to(*this, _fake_type<c10::optional<c10::string_view>>{});
```

Comment:
Ick. You sure you actually want to support this, as opposed to just erroring?

Reply:
I'm not entirely sure what would need to be done in order to avoid calling this overload. So I think it would be best to look into this one in a follow up PR